db/seeds.rb

```diff
@@ -5,3 +5,64 @@
 #   Character.create(name: 'Luke', movie: movies.first)
+
+user_data = [
+  { name: 'admin1', email: 'admin1@example.com', admin: true },
+  { name: 'user1', email: 'user1@example.com', admin: false },
+]
+
+user_data.each do |user|
+  puts "Seeding user: #{user[:name]} and #{user[:email]}"
+  library_user = User.find_or_initialize_by(email: user[:email])
+
+  puts "About to update: #{user[:name]} and #{user[:email]}"
+  library_user.update!(name: user[:name], email: user[:email], admin: user[:admin], password: "password")
+end
+
+book_data = [
+  { title: 'The Golden Compass', author: 'Philip Pullman' },
+  { title: 'The Giver', author: 'Lois Lowry' },
+  { title: 'A Wrinkle in Time', author: "Madeleine L'Engle" },
+  { title: 'To Kill a Mockingbird', author: 'Harper Lee' },
+  { title: 'Lord of the Flies', author: 'William Golding' },
+  { title: 'The Hunger Games', author: 'Suzanne Collins' },
+  { title: 'The Chocolate War', author: 'Robert Cormier' },
+  { title: "Are You There God? It's Me, Margaret" , author: 'Judy Blume' },
+  { title: 'Harry Potter', author: 'J. K. Rowling' },
+  { title: 'The Outsiders', author: 'S. E. Hinton' },
+  { title: 'Sabriel', author: 'Garth Nix' },
+  { title: 'Dealing with Dragons', author: 'Patricia C. Wrede' },
+  { title: 'The Golden Compass', author: 'Philip Pullman' },
+  { title: 'Alanna: The First Adventure', author: 'Tamora Pierce' },
+  { title: 'The Hobbit', author: 'J. R. R. Tolkien' },
+  { title: 'Ella Enchanted', author: 'Gail Carson Levine' },
+  { title: 'Bridge to Terabithia', author: 'Katherine Paterson' },
+  { title: 'Little Women', author: 'Louisa May Alcott' },
+  { title: 'Inkheart', author: 'Cornelia Funke' },
+  { title: 'Charlie and the Chocalate Factory', author: 'Roald Dahl' },
+  { title: 'A Series of Unfortunate Events', author: 'Lemony Snicket' },
+  { title: 'Artemis Fowl', author: 'Eoin Colfer' },
+  { title: 'Tuck Everlasting', author: 'Natalie Babbit' },
+  { title: 'Coraline', author: 'Neil Gaimen' },
+  { title: 'Hatchet', author: 'Gary Paulsen' },
+  { title: 'The Catcher in the Rye', author: 'J. D. Salinger' },
+  { title: 'The Westing Game', author: 'Ellen Raskin' },
+  { title: 'The Princess Bride', author: 'William Goldman' },
+  { title: 'Where the Red Fern Grows', author: 'Wilson Rawls' },
+  { title: 'Walk Two Moons', author: 'Sharon Creech' },
+  { title: 'Holes', author: 'Louis Sachar' },
+  { title: 'Where the Sidewalk Ends', author: 'Shel Silverstein' },
+  { title: 'The True Confessions of Charlotte Doyle', author: 'Avi' },
+  { title: 'Animal Farm', author: 'George Orwell' },
+  { title: 'Redwall', author: 'Brian Jacques' },
+]
+
+book_data.each do |book|
+  puts "Seeding book: #{book[:title]}, #{book[:author]}"
+  library_book = Book.find_or_initialize_by(title: book[:title])
+
+  puts "About to update: #{book[:title]}, #{book[:author]}"
+  library_book.update!(title: book[:title], author: book[:author])
```

Comment:
I'm not sure I understand this. Are there already books in some environments and this adds an author if they book already exists?

Reply:
@paulcsmith, I added `rake db:seed` to my `bin/setup` file. Since it is possible that someone might run `bin/setup` multiple times over the course of a project, I need to ensure that multiple copies of books are not created and that `bin/setup` doesn't fail if run more than once.

Reply:
If `bin/setup` is run again you may want to reset the database. That way you don't need to worry about seeds failing and you always get a clean start. If that doesn't sound good to you you may want to consider `Book.find_or_create_by(title: book[:title], author: book[: author]` and not worry about the updating part.

Also I don't think you need the `puts` for every book. Maybe just one `Creating books` would be enough?

Reply:
Resetting the database seems like a good idea. I will look into how to do that in the `bin/setup` file.